specs/phase0/fork-choice.md

```diff
@@ -439,6 +449,29 @@ def update_unrealized_checkpoints(
         store.unrealized_finalized_checkpoint = unrealized_finalized_checkpoint
 ```
 
+#### `seconds_to_milliseconds`
+
+```python
+def seconds_to_milliseconds(seconds: uint64) -> uint64:
+    """
+    Convert seconds to milliseconds with overflow protection.
+    Returns ``UINT64_MAX`` if the result would overflow.
+    """
+    if seconds > UINT64_MAX // 1000:
+        return UINT64_MAX
+    return seconds * 1000
+```
+
+#### `get_slot_component_duration_ms`
+
+```python
+def get_slot_component_duration_ms(basis_points: uint64) -> uint64:
```

Comment:
```suggestion
def get_slot_component_duration_ms(store: Store, component_id: uint64) -> uint64:
```

I suggest changing this to include the store (so it can detect the fork) and instead of passing the `basis_points`, adding a component id. This way, forks only have to change this function to change component timing

Reply:
I'm not sure that I agree. Clients have convenient ways of handling updated configs/presets; eg a `getAttestationDueBps` function which returns the appropriate value for the current fork. Also, not all slot component times are directly associated with fork choice, like `ATTESTATION_DUE_BPS` in `validator.md`.